incubator/virtualcluster/pkg/syncer/resources/namespace/dws_test.go

```diff
@@ -0,0 +1,301 @@
+	return &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Annotations: map[string]string{
+				constants.LabelUID: uid,
+			},
+		},
+	}
+}
+
+func TestDWNamespaceCreation(t *testing.T) {
+	testTenant := &v1alpha1.Virtualcluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "tenant-1",
+			UID:       "7374a172-c35d-45b1-9c8e-bf5c5b614937",
+		},
+		Spec: v1alpha1.VirtualclusterSpec{},
+		Status: v1alpha1.VirtualclusterStatus{
+			Phase: v1alpha1.ClusterRunning,
+		},
+	}
+
+	defaultNSName := "default"
+	defaultClusterKey := conversion.ToClusterKey(testTenant)
+	defaultSuperNSName := conversion.ToSuperMasterNamespace(defaultClusterKey, defaultNSName)
+
+	testcases := map[string]struct {
+		ExistingObjectInSuper  []runtime.Object
+		ExistingObjectInTenant *v1.Namespace
+
+		ExpectedCreatedNamespace []string
+		ExpectedError            string
+	}{
+		"new namespace": {
+			ExistingObjectInSuper:    []runtime.Object{},
+			ExistingObjectInTenant:   tenantNamespace(defaultNSName, "12345"),
+			ExpectedCreatedNamespace: []string{defaultSuperNSName},
+		},
+		"new namespace but already exists": {
+			ExistingObjectInSuper: []runtime.Object{
+				superNamespace(defaultSuperNSName, "12345"),
+			},
+			ExistingObjectInTenant:   tenantNamespace(defaultNSName, "12345"),
+			ExpectedCreatedNamespace: []string{},
+			ExpectedError:            "",
+		},
+		"new namespace but existing different uid one": {
+			ExistingObjectInSuper: []runtime.Object{
+				superNamespace(defaultSuperNSName, "123456"),
+			},
+			ExistingObjectInTenant:   tenantNamespace(defaultNSName, "12345"),
+			ExpectedCreatedNamespace: []string{},
+			ExpectedError:            "delegated UID is different",
+		},
+	}
+
+	for k, tc := range testcases {
+		t.Run(k, func(t *testing.T) {
+			actions, reconcileErr, err := runDownwardSync(testTenant, tc.ExistingObjectInSuper, tc.ExistingObjectInTenant, tc.ExistingObjectInTenant)
+			if err != nil {
+				t.Errorf("%s: error running downward sync: %v", k, err)
+				return
+			}
+
+			if reconcileErr != nil {
+				if tc.ExpectedError == "" {
+					t.Errorf("expected no error, but got \"%v\"", err)
+				} else if !strings.Contains(reconcileErr.Error(), tc.ExpectedError) {
+					t.Errorf("expected error msg \"%s\", but got \"%v\"", tc.ExpectedError, err)
+				}
+			} else {
+				if tc.ExpectedError != "" {
+					t.Errorf("expected error msg \"%s\", but got empty", tc.ExpectedError)
+				}
+			}
+
+			if len(tc.ExpectedCreatedNamespace) != len(actions) {
+				t.Errorf("%s: Expected to create namespace %#v. Actual actions were: %#v", k, tc.ExpectedCreatedNamespace, actions)
+				return
+			}
+			for i, expectedName := range tc.ExpectedCreatedNamespace {
+				action := actions[i]
+				if !action.Matches("create", "namespaces") {
+					t.Errorf("%s: Unexpected action %s", k, action)
+				}
+				createdNS := action.(core.CreateAction).GetObject().(*v1.Namespace)
```

Comment:
simply check the creation action. The spec checking should be done in the UT for package `conversion`